src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py

```diff
@@ -423,4 +427,22 @@ def title(self):
         Check DataExplorer.py, line 355
         """
         return "Corfunc Perspective"
+
+    def on_save(self):
+        """
+        Save corfunc state into a file
+        """
+        f_name = QtWidgets.QFileDialog.getSaveFileName(
+            caption="Save As",
+            filter="Corfunc Text Output (*.crf)",
+            parent=None)[0]
+        if "." not in f_name:
+            f_name += ".crf"
```

Comment:
When cancelled the dialog, this will create a file called ".crf" and save the data to it. Probably not what expected.